### ISSUE TITLE
task: invalid choice: 'speech_dlm_task'

### ISSUE THREAD

**Issue by HelenJi-maker** [facebookresearch/fairseq#5522]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
I try the task textless_nlp/dglsm training，but  report an error ，task: invalid choice: 'speech_dlm_task'

#### Code
cd fairseq_cli
python train.py binary_dir --save-dir ./checkpoint --tensorboard-logdir ./checkpoint --task speech_dlm_task --channels unitA,unitB --next-unit-prediction "False" --edge-unit-prediction "True" --duration-prediction "True" --delayed-duration-target "True" --criterion speech_dlm_criterion --arch speech_dlm --decoder-cross-layers 4 --share-decoder-input-output-embed --dropout 0.1 --attention-dropout 0.1 --optimizer adam --adam-betas "(0.9, 0.98)" --clip-norm 1.0 --lr 0.0005 --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07 --max-tokens 18432 --tokens-per-sample 6144 --sample-break-mode none --update-freq 16 --num-workers 4 --skip-invalid-size-inputs-valid-test --max-update 250000 --warmup-updates 20000 --save-interval-updates 10000 --keep-last-epochs 1 --no-epoch-checkpoints --log-interval 50 --seed 100501 --fp16 --checkpoint-activations

the error is :
train.py: error: argument --task: invalid choice: 'speech_dlm_task' (choose from 'audio_pretraining', 'audio_finetuning', 'cross_lingual_lm', 'denoising', 'speech_to_text', 'text_to_speech', 'frm_text_to_speech', 'hubert_pretraining', 'language_modeling', 'legacy_masked_lm', 'masked_lm', 'multilingual_denoising', 'multilingual_language_modeling', 'multilingual_masked_lm', 'speech_unit_modeling', 'translation', 'multilingual_translation', 'online_backtranslation', 'semisupervised_translation', 'sentence_prediction', 'sentence_prediction_adapters', 'sentence_ranking', 'simul_speech_to_text', 'simul_text_to_text', 'speech_to_speech', 'translation_from_pretrained_bart', 'translation_from_pretrained_xlm', 'translation_lev', 'translation_multi_simple_epoch', 'dummy_lm', 'dummy_masked_lm', 'dummy_mt')

#### What have you tried?
I have try the train instruction，
![image](https://github.com/user-attachments/assets/248df24c-46a4-4eac-ba8b-df5203108247)

#### What's your environment?

 - fairseq Version (e.g., 1.0 or main):  0.12.2
 - PyTorch Version (e.g., 1.0)  1.13.0+cu117
 - OS (e.g., Linux):  linux
 - How you installed fairseq (`pip`, source): pip install fairseq
 - Build command you used (if compiling from source):
 - Python version: 3.8.9
 - CUDA/cuDNN version: 11.4
 - GPU models and configuration:
 - Any other relevant information:
